npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -64,6 +65,22 @@ def instrument_package_manager_version
         )
       end
 
+      def yarn_version
+        @yarn_version ||= begin
+          package = JSON.parse(package_json.content)
+          if (pkgmanager = package.fetch("packageManager", nil))
+            @yarn_version = get_yarn_version_from_path(pkgmanager)
+          elsif yarn_lock
+            @yarn_version = 1
+          end
+        end
```

Comment:
Noting that this memoization pattern is prone to causing re-evaluation since `if`/`elseif` will fall back to `nil`. We could fall back to `return @yarn_version if defined?(@yarn_version)`.

Also noting that we don't need the `@yarn_version` assignments inside the `if` block.